From: Yehuda Sadeh <yehuda@redhat.com>
Date: Thu, 30 Jun 2016 17:34:38 -0700
Subject: rgw: set args when rebuilding req_info

Was missing args, so it didn't sign requests with subresources
correctly when forwarding.

Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>
(cherry picked from commit dd1068c54e50671d8904b99189823418a5a5bc07)

Resolves (with following): rhbz#1350522
---
 src/rgw/rgw_common.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/rgw/rgw_common.cc b/src/rgw/rgw_common.cc
index d309015..f430b2e 100644
--- a/src/rgw/rgw_common.cc
+++ b/src/rgw/rgw_common.cc
@@ -168,6 +168,7 @@ void req_info::rebuild_from(req_info& src)
 {
   method = src.method;
   script_uri = src.script_uri;
+  args = src.args;
   if (src.effective_uri.empty()) {
     request_uri = src.request_uri;
   } else {
